airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlCursorBasedIntegrationTest.kt

```diff
@@ -0,0 +1,129 @@
+import io.airbyte.cdk.discover.JdbcAirbyteStreamFactory
+import io.airbyte.cdk.jdbc.IntFieldType
+import io.airbyte.cdk.jdbc.JdbcConnectionFactory
+import io.airbyte.cdk.jdbc.StringFieldType
+import io.airbyte.cdk.output.BufferingOutputConsumer
+import io.airbyte.integrations.source.mysql.MysqlContainerFactory.execAsRoot
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage
+import io.airbyte.protocol.models.v0.AirbyteStateMessage
+import io.airbyte.protocol.models.v0.AirbyteStream
+import io.airbyte.protocol.models.v0.CatalogHelpers
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteStream
+import io.airbyte.protocol.models.v0.StreamDescriptor
+import io.airbyte.protocol.models.v0.SyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.sql.Connection
+import java.sql.Statement
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.Timeout
+import org.testcontainers.containers.MySQLContainer
+
+class MysqlCursorBasedIntegrationTest {
+
+    @Test
+    fun testCursorBasedRead() {
+        val run1: BufferingOutputConsumer =
+            CliRunner.source("read", config(), configuredCatalog).run()
+
+        val lastStateMessageFromRun1 = run1.states().last()
+        val lastStreamStateFromRun1 = lastStateMessageFromRun1.stream.streamState
+
+        assertEquals("2", lastStreamStateFromRun1.get("cursor").asText())
+        assertEquals(2, lastStreamStateFromRun1.get("version").asInt())
+        assertEquals("cursor_based", lastStreamStateFromRun1.get("state_type").asText())
+        assertEquals("tbl", lastStreamStateFromRun1.get("stream_name").asText())
+        assertEquals(listOf("k"), lastStreamStateFromRun1.get("cursor_field").map { it.asText() })
+        assertEquals("test", lastStreamStateFromRun1.get("stream_namespace").asText())
+        assertEquals(0, lastStreamStateFromRun1.get("cursor_record_count").asInt())
+
+        connectionFactory.get().use { connection: Connection ->
+            connection.isReadOnly = false
+            connection.createStatement().use { stmt: Statement ->
+                stmt.execute("INSERT INTO test.tbl (k, v) VALUES (3, 'baz')")
+            }
+        }
+
+        val run2InputState: List<AirbyteStateMessage> = listOf(lastStateMessageFromRun1)
+        val run2: BufferingOutputConsumer =
+            CliRunner.source("read", config(), configuredCatalog, run2InputState).run()
+        val recordMessageFromRun2: List<AirbyteRecordMessage> = run2.records()
+        assertEquals(recordMessageFromRun2.size, 1)
+    }
+
+    companion object {
+        val log = KotlinLogging.logger {}
+        lateinit var dbContainer: MySQLContainer<*>
+
+        fun config(): MysqlSourceConfigurationJsonObject =
+            MysqlContainerFactory.config(dbContainer).apply {
+                setCursorMethodValue(UserDefinedCursor)
```

Comment:
you can remove this now that you've added it by default in the MysqlContainerFactory